services/spar/src/Spar/API.hs

```diff
@@ -375,7 +375,7 @@ validateNewIdP apiversion _idpMetadata teamId mReplaces = withDebugLog "validate
     Data.GetIdPFound idp' {- same team -} -> handleIdPClash (Right idp')
     Data.GetIdPNotFound -> pure ()
     res@(Data.GetIdPDanglingId _) -> throwSpar . SparIdPNotFound . ("validateNewIdP: " <>) . cs . show $ res -- database inconsistency
-    res@(Data.GetIdPNonUnique _) -> throwSpar . SparIdPNotFound . ("validateNewIdP: " <>) . cs . show $ res -- impossible
+    Data.GetIdPNonUnique ids' {- same team didn't yield anything, but there are at least two other teams with this issuer already -} -> handleIdPClash (Left ids')
```

Comment:
It seems unnecessary to distinguish between the "non unique" case and the "wrong team", since "non unique" implies "wrong team", and they can still be distinguished by the size of the list.

Reply:
Good idea, but separate PR.  (And I think this may become more obvious of a change if I make the CRUD API for IdPs in `Spar.Data` less abstract, and expose functions that do precisely what's needed on the call side, and nothing else.)